crates/services/consensus_module/poa/src/verifier.rs

```diff
@@ -0,0 +1,140 @@
+    fn block_header(&self, height: &BlockHeight) -> StorageResult<BlockHeader>;
+
+    /// Gets the block header BMT MMR root at `height`.
+    fn block_header_merkle_root(&self, height: &BlockHeight) -> StorageResult<Bytes32>;
+}
+
+pub fn verify_poa_block_fields<D: Database>(
+    config: &Config,
+    database: &D,
+    block: &Block,
+) -> anyhow::Result<()> {
+    let height = *block.header().height();
+    ensure!(
+        height != 0u32.into(),
+        "The PoA block can't have the zero height"
+    );
+
+    let prev_height = height - 1u32.into();
+    let prev_root = database.block_header_merkle_root(&prev_height)?;
+    let header = block.header();
+    ensure!(
+        header.prev_root() == &prev_root,
+        "The genesis time should be unix epoch time"
+    );
+
+    let prev_header = database.block_header(&prev_height)?;
+
+    ensure!(
+        header.da_height >= prev_header.da_height,
+        "The `da_height` of the next block can't be lower"
+    );
+
+    // Skip the verification of the time if it is possible to produce blocks manually.
+    if !config.enabled_manual_blocks {
+        if config.perform_strict_time_rules {
+            match config.production {
+                PoABlockProduction::Instant => {
+                    ensure!(
+                        header.time() >= prev_header.time(),
+                        "The `time` of the next block can't be lower"
+                    );
+                }
+                PoABlockProduction::Interval {
+                    block_time: average_block_time,
+                } => {
+                    let previous_block_time = Tai64N::from(prev_header.time());
+
+                    // If the `average_block_time` is 15 seconds, the block should be in the range
+                    // [15..15 + 15/2] -> [15..23]
+                    let average_block_time = average_block_time;
+                    let half_average_block_time = average_block_time / 2;
+                    let lower_bound = previous_block_time.add(average_block_time);
+                    let upper_bound = previous_block_time
+                        .add(average_block_time + half_average_block_time);
+                    let block_time = Tai64N::from(header.time());
+                    ensure!(
+                        lower_bound <= block_time && block_time <= upper_bound,
+                        "The `time` of the next should be more than {:?} but less than {:?} but got {:?}",
+                        lower_bound,
+                        upper_bound,
+                        block_time,
+                    );
+                }
+                PoABlockProduction::Hybrid {
+                    min_block_time,
+                    max_block_time,
+                    ..
+                } => {
+                    let previous_block_time = Tai64N::from(prev_header.time());
+                    // The block should be in the range
+                    // [min..max + (max - min)/2]
```

Comment:
I don't understand why this is true?

Reply:
Because it is not true=) Created an issue to discuss it https://github.com/FuelLabs/fuel-core/issues/918

I already disabled this check in the production for now.